From 4aed9c0b2ae244f1a57337ff8c3dc6070c087c49 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Diego=20'Flameeyes'=20Petten=C3=B2?= <flameeyes@gmail.com>
Date: Thu, 13 Nov 2008 01:23:35 +0100
Subject: [PATCH] Allow opt-out from jack, pulseaudio and avcodec dependencies.

Without this patch the jack, pulseaudio and avcodec discovery was
"automagic", without a way for the user to disable the relative
plugins if the dependencies are installed but the plugin is unwanted.

This patch does not change the default behaviour but allows to opt-out
from the plugins by passing the relative --without option at
./configure time.
---
 configure.in |   31 ++++++++++++++++++++++++++-----
 1 files changed, 26 insertions(+), 5 deletions(-)

diff --git a/configure.in b/configure.in
index 9efb890..ffb6528 100644
--- a/configure.in
+++ b/configure.in
@@ -15,13 +15,28 @@ PKG_CHECK_MODULES(ALSA, alsa >= 1.0.11)
 AC_CHECK_LIB(asound, snd_pcm_ioplug_create,,
 	     AC_ERROR([*** libasound has no external plugin SDK]), -ldl)
 
-PKG_CHECK_MODULES(JACK, jack >= 0.98, [HAVE_JACK=yes], [HAVE_JACK=no])
+AC_ARG_WITH([jack],
+      AS_HELP_STRING([--without-jack], [Disable building of JACK plugin]))
+
+if test "x$with_jack" != "xno"; then
+  PKG_CHECK_MODULES(JACK, jack >= 0.98, [HAVE_JACK=yes], [HAVE_JACK=no])
+fi
 AM_CONDITIONAL(HAVE_JACK, test x$HAVE_JACK = xyes)
 
-PKG_CHECK_MODULES(pulseaudio, [libpulse >= 0.9.2], [HAVE_PULSE=yes], [HAVE_PULSE=no])
+AC_ARG_WITH([pulseaudio],
+      AS_HELP_STRING([--without-pulseaudio], [Disable building of pulseaudio plugin]))
+
+if test "x$with_pulseaudio" != "xno"; then
+  PKG_CHECK_MODULES(pulseaudio, [libpulse >= 0.9.2], [HAVE_PULSE=yes], [HAVE_PULSE=no])
+fi
 AM_CONDITIONAL(HAVE_PULSE, test x$HAVE_PULSE = xyes)
 
-PKG_CHECK_MODULES(samplerate, [samplerate], [HAVE_SAMPLERATE=yes], [HAVE_SAMPLERATE=no])
+AC_ARG_WITH([samplerate],
+      AS_HELP_STRING([--without-samplerate], [Disable building of samplerate plugin]))
+
+if test "x$with_samplerate" != "xno"; then
+  PKG_CHECK_MODULES(samplerate, [samplerate], [HAVE_SAMPLERATE=yes], [HAVE_SAMPLERATE=no])
+fi
 AM_CONDITIONAL(HAVE_SAMPLERATE, test x$HAVE_SAMPLERATE = xyes)
 
 AC_ARG_ENABLE([maemo-plugin],
@@ -45,8 +60,14 @@ if test "$use_maemo_rm" = "yes"; then
         fi
 fi
 
-PKG_CHECK_MODULES(AVCODEC, [libavcodec], [HAVE_AVCODEC=yes], [HAVE_AVCODEC=no])
-if test "$HAVE_AVCODEC" != "yes"; then
+AC_ARG_WITH([avcodec],
+      AS_HELP_STRING([--without-avcodec], [Don't build plugins depending on avcodec (a52)]))
+
+if test "x$with_avcodec" != "xno"; then
+  PKG_CHECK_MODULES(AVCODEC, [libavcodec], [HAVE_AVCODEC=yes], [HAVE_AVCODEC=no])
+fi
+
+if test "$HAVE_AVCODEC" = "no"; then
   AC_ARG_WITH([avcodec-includedir],
 	AS_HELP_STRING([--with-avcodec-includedir=dir],
 		[AVcodec include directory]),
-- 
1.6.0.3

